### PR TITLE
feat(Twitch-Notifications): Added live-Roles

### PR DESCRIPTION
**Pull request by jateute**

### CODE REVIEW

File: modules/twitch-notifications/configs/streamers.json [hunk 1]
```diff
@@ -79,6 +79,26 @@
       "type": "string",
       "description-en": "Streamer where a notification should send when they start streaming",
       "description-de": "Steamer, bei denen eine Benachrichtigung gesendet werden soll, wenn sie anfangen, zu streamen"
+    },
+    {
+      "field_name": "id",
+      "humanname-en": "Discord-User ID",
+      "humanname-de": "Discord-Benutzer ID",
+      "default": "",
+      "type": "string",
```
Comment (SCDerox) [inline modules/twitch-notifications/configs/streamers.json:88]: If you want, we can merge #76 before this so the new type is supported.
```suggestion
      "type": "userID",
```

**Review by SCDerox — COMMENTED** [review]
Haven't tested it yet, but I have a recommendation to improve user experience:
Adding a new boolean-field, which enables the life-role-functionality and uses the `dependsOn` field. This way, the userID and role field will be hidden, until the boolean field gets enabled (see #76). Here a small demo: https://user-images.githubusercontent.com/43812987/195059667-c6d6d033-ecef-47be-8d52-f799565b350b.mp4
Let me know what you think ^^

File: modules/twitch-notifications/configs/streamers.json (hunk 1 above)
Comment (jateute) [reply modules/twitch-notifications/configs/streamers.json:88]: Yeah, would be great

Comment (SCDerox) [reply modules/twitch-notifications/configs/streamers.json:88]: Merged :+1:

File: modules/twitch-notifications/events/botReady.js [hunk 2]
```diff
@@ -75,10 +93,24 @@ function twitchNotifications(client, apiClient) {
                 startedAt: stream.startDate.toString()
             });
             sendMsg(stream.userDisplayName, stream.gameName, stream.thumbnailUrl, streamers[index]['liveMessageChannel'], stream.title, index);
+            addLiveRole(streamers[index]['id'], streamers[index]['role']);
```
Comment (SCDerox) [inline modules/twitch-notifications/events/botReady.js:96]:
```suggestion
            addLiveRole(streamers[index]['id'], streamers[index]['role'], streamers[index]['liveRole']);
```

File: modules/twitch-notifications/events/botReady.js [hunk 3]
```diff
@@ -75,10 +93,24 @@ function twitchNotifications(client, apiClient) {
...
         } else if (stream !== null && stream.startDate.toString() !== streamer.startedAt) {
             streamer.startedAt = stream.startDate.toString();
             streamer.save();
             sendMsg(stream.userDisplayName, stream.gameName, stream.thumbnailUrl, streamers[index]['liveMessageChannel'], stream.title, index);
+            addLiveRole(streamers[index]['id'], streamers[index]['role']);
```
Comment (SCDerox) [inline modules/twitch-notifications/events/botReady.js:101]:
```suggestion
            addLiveRole(streamers[index]['id'], streamers[index]['role'], streamers[index]['liveRole']);
```

**Review by SCDerox — CHANGES_REQUESTED** [review]
This does not actually work because you are not passing the config value to your own function. If this get's fixed, everything works as intended.

**Review by SCDerox — APPROVED** [review]
lgtm

**Pull request merged**